pos_payment_terminal/static/src/xml/pos_payment_terminal.xml

```diff
@@ -3,11 +3,29 @@
     <t t-extend="PaymentScreen-Paymentlines" >
         <t t-jquery="th:last" t-operation="before">
             <th name="terminal-transaction"></th>
+            <th name="terminal-transaction-spinner"></th>
+            <th name="terminal-transaction-status"></th>
         </t>
         <t t-jquery="td.delete-button" t-operation="before">
             <t t-if="line.cashregister.journal.payment_mode and widget.pos.config.iface_payment_terminal">
                 <td>
-                   <button class="payment-terminal-transaction-start"  t-att-data-cid='line.cid'>Start transaction</button>
+                   <button t-if="!(line.terminal_transaction_id and line.terminal_transaction_success === null) and line.terminal_transaction_success !== true" class="payment-terminal-transaction-start" t-att-data-cid='line.cid'>Start transaction</button>
+                </td>
+                <td>
+                   <img t-if="line.terminal_transaction_id and line.terminal_transaction_success === null" class="payment-spinner" src="/web/static/src/img/throbber.gif"/>
+                </td>
+                 <td class="payment-status">
+                   <t class="payment-status" t-if="line.terminal_transaction_success !== undefined and line.terminal_transaction_status">
+                       <span t-esc="line.terminal_transaction_status"/>
+                   </t>
+                </td>
```

Comment:
add properties for visibility of each UI component

Reply:
done